library/core/src/str/mod.rs

```diff
@@ -1108,7 +1108,7 @@ impl str {
         LinesAny(self.lines())
     }
 
-    /// Returns an iterator of `u16` over the string encoded as UTF-16.
+    /// Returns an iterator of `u16` over the string encoded as native endian UTF-16.
```

Comment:
This should probably mention the BOM detail in some sort of fashion, like "if you want a BOM, add it yourself?"